from typing import Union

from .hola_pb2 import *

GRPCEvent = Union[
    FriendRequestEvent,
    GroupRequestEvent,
    PrivateMessageEvent,
    GroupMessageEvent,
    FriendAddNoticeEvent,
    FriendRecallNoticeEvent,
    GroupIncreaseNoticeEvent,
    GroupDecreaseNoticeEvent,
    GroupRecallNoticeEvent,
    GroupBanNoticeEvent,
    GroupAdminNoticeEvent,
    PokeNotifyEvent,
]

GRPCRequest = Union[
    SendMessageRequest,
    DeleteMessageRequest,
    GetMessageRequest,
    GetForwardMessageRequest,
    SendLikeRequest,
    SetGroupKickRequest,
    SetGroupBanRequest,
    SetGroupAnonymousBanRequest,
    SetGroupWholeBanRequest,
    SetGroupAdminRequest,
    SetGroupAnonymousRequest,
    SetGroupCardRequest,
    SetGroupNameRequest,
    SetGroupLeaveRequest,
    SetGroupSpecialTitleRequest,
    SetFriendAddRequestRequest,
    SetGroupAddRequestRequest,
    GetSelfInfoRequest,
    GetUserInfoRequest,
    GetFriendListRequest,
    GetGroupInfoRequest,
    GetGroupListRequest,
    GetGroupMemberInfoRequest,
    GetGroupMemberListRequest,
    SendForwardMessageRequest
]

GRPCResult = Union[
    SendMessageResult,
    DeleteMessageResult,
    GetMessageResult,
    GetForwardMessageResult,
    SendLikeResult,
    SetGroupKickResult,
    SetGroupBanResult,
    SetGroupAnonymousBanResult,
    SetGroupWholeBanResult,
    SetGroupAdminResult,
    SetGroupAnonymousResult,
    SetGroupCardResult,
    SetGroupNameResult,
    SetGroupLeaveResult,
    SetGroupSpecialTitleResult,
    SetFriendAddRequestResult,
    SetGroupAddRequestResult,
    GetSelfInfoResult,
    GetUserInfoResult,
    GetFriendListResult,
    GetGroupInfoResult,
    GetGroupListResult,
    GetGroupMemberInfoResult,
    GetGroupMemberListResult,
    SendForwardMessageResult
]

GRPCReceivableMessageSegment = Union[
    TextSegment,
    AtSegment,
    FaceSegment,
    RpsSegment,
    DiceSegment,
    ImageSegment,
    RecordSegment,
    VideoSegment,
    LocationSegment,
    ShareSegment,
    JsonSegment,
    XmlSegment,
    PokeSegment,
    ReplySegment,
]

GRPCSendibleMessageSegment = Union[
    TextSegment,
    FaceSegment,
    ImageSegment,
    RecordSegment,
    VideoSegment,
    AtSegment,
    RpsSegment,
    DiceSegment,
    PokeSegment,
    AnonymousSegment,
    ShareSegment,
    ContactSegment,
    LocationSegment,
    MusicSegment,
    CustomMusicSegment,
    ReplySegment,
    XmlSegment,
    JsonSegment,
]
